File: back-end-nodejs/src/models/cartModel.js
import mongoose from 'mongoose';
import { OrderItemType } from './enums.js';

// ========== Cart Item (embedded) ==========
const cartItemSchema = new mongoose.Schema(
  {
    type: {
      type: String,
      enum: Object.values(OrderItemType),
      default: OrderItemType.READY_MADE,
    },
    giftBoxId: { type: String, default: null },
    customBoxId: { type: String, default: null },
    quantity: { type: Number, default: 0 },
    unitPrice: { type: Number, default: 0 },
    addedAt: { type: Date, default: Date.now },
  },
  { _id: true }
);

// ========== Cart Model ==========
const cartSchema = new mongoose.Schema(
  {
    userId: { type: String, default: null },
    sessionId: { type: String, default: null },
    items: [cartItemSchema],
  },
  { timestamps: true }
);

export const Cart = mongoose.model('Cart', cartSchema);

File: back-end/ShopHangTet/Models/CartModel.cs
using MongoDB.Bson;
using MongoDB.Bson.Serialization.Attributes;

namespace ShopHangTet.Models
{
    /// Cart Model - Giỏ hàng (Guest & Member)
    public class Cart
    {
        [BsonId]
        [BsonRepresentation(BsonType.ObjectId)]
        public string Id { get; set; } = ObjectId.GenerateNewId().ToString();

        [BsonElement("userId")]
        public string? UserId { get; set; } // Nullable for Guest

        [BsonElement("sessionId")]
        public string? SessionId { get; set; } // For Guest checkout

        [BsonElement("items")]
        public List<CartItem> Items { get; set; } = new();

        [BsonElement("createdAt")]
        public DateTime CreatedAt { get; set; } = DateTime.UtcNow;

        [BsonElement("updatedAt")]
        public DateTime UpdatedAt { get; set; } = DateTime.UtcNow;
    }

    public class CartItem
    {
        [BsonId]
        [BsonRepresentation(BsonType.ObjectId)]
        public string Id { get; set; } = ObjectId.GenerateNewId().ToString();

        public string CartId { get; set; } = string.Empty;

        [BsonElement("userId")]
        public string? UserId { get; set; } 

        [BsonElement("sessionId")]
        public string? SessionId { get; set; }

        [BsonElement("type")]
        public OrderItemType Type { get; set; } = OrderItemType.READY_MADE; // READY_MADE, MIX_MATCH

        [BsonElement("giftBoxId")]
        public string? GiftBoxId { get; set; }

        [BsonElement("customBoxId")]
        public string? CustomBoxId { get; set; }

        [BsonElement("quantity")]
        public int Quantity { get; set; }

        [BsonElement("unitPrice")]
        public decimal UnitPrice { get; set; }

        [BsonElement("addedAt")]
        public DateTime AddedAt { get; set; } = DateTime.UtcNow;
    }
}